src/main/kotlin/com/sourcegraph/cody/chat/ui/ContextFilesPanel.kt

```diff
@@ -40,67 +43,148 @@ class ContextFilesPanel(
+      }
+      filesAvailableInWeb.forEach { file: Pair<RemoteContextFile, ContextFileFile> ->
+        val filePanel = createRemoteFileWithLinkPanel(file.first, file.second)
+        accordionSection.contentPanel.add(filePanel)
       }
       add(accordionSection, BorderLayout.CENTER)
     }
   }
 
   @RequiresEdt
-  private fun createFileWithLinkPanel(file: VirtualFile): JPanel {
-    val projectRelativeFilePath = file.path.removePrefix(project.basePath ?: "")
-    val anAction =
+  private fun createLocalFileWithLinkPanel(
+      file: VirtualFile,
+      contextFileFile: ContextFileFile?
+  ): JPanel {
+    val displayTextWithoutPrefix = file.path.removePrefix(project.basePath ?: "")
+    val actionText =
+        getActionTextForFileLinkAction(contextFileFile, LocalContextFile(displayTextWithoutPrefix))
+    return createLinkPanel(actionText) {
+      object : DumbAwareAction() {
+        override fun actionPerformed(anActionEvent: AnActionEvent) {
+          logger.info(
+              "Opening a file from the used context (displayText=$displayTextWithoutPrefix, file=$file)")
+          FileEditorManager.getInstance(project).openFile(file, /*focusEditor=*/ true)
+        }
+      }
+    }
+  }
+
+  @RequiresEdt
+  private fun createRemoteFileWithLinkPanel(
+      remoteContextFile: RemoteContextFile,
+      contextFileFile: ContextFileFile
+  ) =
+      createLinkPanel(getActionTextForFileLinkAction(contextFileFile, remoteContextFile)) {
         object : DumbAwareAction() {
           override fun actionPerformed(anActionEvent: AnActionEvent) {
             logger.info(
-                "Opening a file from the used context (projectRelativeFilePath=$projectRelativeFilePath, file=$file)")
-            FileEditorManager.getInstance(project).openFile(file, /*focusEditor=*/ true)
+                "Opening a file from the used context (displayText=${remoteContextFile.displayText})")
+            openInBrowser(project, contextFileFile.uri.path)
           }
         }
-    val goToFile = AnActionLink("@$projectRelativeFilePath", anAction)
+      }
+
+  private fun createLinkPanel(actionText: String, anAction: () -> AnAction): JPanel {
+    val goToFile = AnActionLink(actionText, anAction.invoke())
+    goToFile.toolTipText = actionText
     val panel = JPanel(BorderLayout())
     panel.isOpaque = false
     panel.border = JBUI.Borders.emptyLeft(3)
     panel.add(goToFile, BorderLayout.PAGE_START)
     return panel
   }
 
+  private fun getActionTextForFileLinkAction(
+      contextFileFile: ContextFileFile?,
+      typedContextFile: TypedContextFile
+  ): String {
+    val startLine = contextFileFile?.range?.start?.line
+    val endLine = contextFileFile?.range?.end?.line
+    return buildString {
+      if (typedContextFile is LocalContextFile) {
+        append("@")
+      }
+      append(typedContextFile)
+      if (startLine != null && endLine != null) {
+        if (startLine != endLine) {
+          append(":$startLine-$endLine")
```

Comment:
What if `startLine == endLine`? Don't we want to append `:$startLine` in such case?